src/hw/actions/shared.js

```diff
@@ -0,0 +1,29 @@
+// @flow
+
+import { ReplaySubject } from "rxjs";
+import { useEffect, useState } from "react";
+import type { DeviceModelId } from "@ledgerhq/devices";
+
+export type Device = {
+  path: string,
+  modelId: DeviceModelId
+};
+
+export type Action<Request, HookState, Result> = {|
+  useHook: (?Device, Request) => HookState,
+  mapResult: HookState => ?Result
+|};
+
+// emit value each time it changes by reference. it replays the last value at subscribe time.
+export function useReplaySubject<T>(value: T): ReplaySubject<T> {
```

Comment:
could we move this into a src/observable.js maybe ?

Reply:
ok